frontend/lib/norent/letter-content.tsx

```diff
@@ -168,34 +166,66 @@ export const NorentLetterEmailForUserStaticPage: React.FC<{}> = () => (
 
 export const NorentLetterContent: React.FC<NorentLetterContentProps> = (
   props
-) => (
-  <>
-    <LetterTitle {...props} />
-    <p className="has-text-right">{friendlyDate(new Date())}</p>
-    <LetterHeading {...props} />
-    <p>
-      Dear <LandlordName {...props} />,
-    </p>
-    <p>
-      This letter is to notify you that I will be unable to pay rent starting on{" "}
-      <PaymentDate {...props} /> and until further notice due to loss of income,
-      increased expenses, and/or other financial circumstances related to
-      COVID-19.
-    </p>
-    <TenantProtections {...props} />
-    <p>
-      In order to document our communication and to avoid misunderstandings,
-      please reply to me via email or text rather than a call or visit.
-    </p>
-    <p>Thank you for your understanding and cooperation.</p>
-    <p className="jf-signature">
-      Regards,
-      <br />
-      <br />
-      <FullName {...props} />
-    </p>
-  </>
-);
+) => {
+  const state = props.state as USStateChoice;
+  const letterVersion = getNorentMetadataForUSState(state).lawForLetter
+    .whichVersion;
+  return (
+    <>
+      <LetterTitle {...props} />
+      <p className="has-text-right">{friendlyDate(new Date())}</p>
+      <LetterHeading {...props} />
+      <p>
+        Dear <LandlordName {...props} />,
+      </p>
+      {letterVersion === CovidStateLawVersion.V1_NON_PAYMENT ? (
```

Comment:
This is sort of type nerdery, but if you move the version-specific content into a separate function that uses a `switch` statement, you can use TypeScript's [exhaustiveness checking](https://www.typescriptlang.org/docs/handbook/advanced-types.html#exhaustiveness-checking) so that if we ever decide to add another "legal version type" to our Enum, the source code will fail to typecheck because we haven't dealt with the new version in our switch statement!  It's kinda neat in that u can make sure you're dealing with all the edge cases without having to even run the test suite.

Aannnnyhoo that's obviously for another day when we have a lot more time on our hands!

Reply:
hahahah definitely. Tbh I hate switch statements... they're so syntactically verbose!